### PR TITLE
add glossary

### PR DESCRIPTION
**Pull request by mschrimpf**
with @gretatuckute

### CODE REVIEW

**Comment by gretatuckute** [comment]
Happy to approve but as we wrote it together maybe @kvfairchild wants to take a quick look :)

**Comment by kvfairchild** [comment]
Really nice!

It looks like this glossary is written at the `core` level and all links are to `core` documentation... should this potentially be added to `core` instead and linked to from `language`?

**Comment by gretatuckute** [comment]
> yea I am okay with moving this over to core (I had initially thought this would end up being more language-specific, but it's pretty generic). I would want to add e.g. electrophysiology to the neural recording. We can then also link to the glossary from the language docs.
> 
> @gretatuckute what do you think?

I think it'd be great to have this as part of the core. Let's just link to from the frontpage of the language documentation as well as the GitHub readme?

File: docs/source/modules/glossary.rst [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        :doc:`ArtificialSubject <./artificial_subject>`
+
+    Benchmark
+        A benchmark runs an experiment on an ArtificialSubject,
+        and compares the resulting measurements (often model predictions) to biological measurements using a particular metric,
+        resulting in a similarity score.
+        :doc:`Benchmarks <https://brain-score-core.readthedocs.io/en/latest/modules/_autosummary/brainscore_core.benchmarks.html#module-brainscore_core.benchmarks>`
+
+    Plug-in
+        Plug-ins are ways of flexibly adding new data, metrics, benchmarks, or models to the Brain-Score platform.
```
Comment (kvfairchild) [inline docs/source/modules/glossary.rst:43]: More context here would be useful, maybe adding something like

New data, metrics, benchmarks, and models can be added to the Brain-Score platform as standalone modules called "plug-ins". Plug-ins are designed to be easily submitted by the community and can have their own set of dependencies beyond those included in the main Brain-Score codebase. << insert link(s) to relevant plugin documentation here >>

File: docs/source/modules/glossary.rst [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+    Xarray
+        `Xarray <https://docs.xarray.dev/en/stable/>` DataArrays are multidimensional pandas dataframes.
+        DataArrays enable us to retain metadata along several dimensions of the data at once (often needed in neuroscience data, e.g., keeping track of stimuli and neural dimensions).
+
+    Target
+        A target is a term for the data that is to-be predicted in regression-based metrics;
```
Comment (kvfairchild) [inline docs/source/modules/glossary.rst:50]:
```suggestion
        A target is a term for the data that is to be predicted in regression-based metrics;
```

File: docs/source/modules/glossary.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        An ArtificialSubject implements a set of interface functions that benchmarks can interact with.
+        This way, we can run experiments on computational models implementing these functions
+        in the same way as a human experimental subject. These interface functions include
+        behavioral tasks (e.g., next-word prediction or reading times),
+        neural recordings (e.g. fMRI or ECoG recordings in the biological brain which neural network models could implement as layer-wise unit activations),
+        and a method to digest stimuli.
+        :doc:`ArtificialSubject <./artificial_subject>`
+
+    Benchmark
+        A benchmark runs an experiment on an ArtificialSubject,
```
Comment (kvfairchild) [inline docs/source/modules/glossary.rst:37]: Maybe explicitly state here that each benchmark incorporates a dataset and a metric?

**Review by kvfairchild — APPROVED** [review]

File: docs/source/modules/glossary.rst (hunk 1 above)
Comment (mschrimpf) [reply docs/source/modules/glossary.rst:43]: I'm committing the change, but don't know where to point to for documentation

**Pull request merged**